ONSdigital/response-operations-ui
# Role based access messages

# What and why?
Role-based access needed to be implemented for the secure message-related pages. The functionality to reply to messages, as well as re-open/close conversations is now limited to users who have the `message.edit` permission.

Additionally, pipenv dependencies have been updated to fix build issues, and `black` has been bumped to version `22.3.0` in order to fix an issue that was occurring with the `click` dependency. You can read more about this issue [here](https://github.com/psf/black/issues/2964).

# How to test?
* Deploy to your environment.
* Run acceptance tests.
* Go to R-ops and click on the reporting units tab, find an RU and click on a survey for that RU. If role-based access is enabled (you can toggle this by editing the deployment config), you should see the button to send messages. If not, the button shouldn't appear.
* Go to the secure messages tab and find a conversation. You should see the links/buttons to assign a survey, reply to a message, and close conversation when role-based access is enabled. Likewise, you shouldn't see these when it's disabled.
* Click the button to close the conversation. On the next page, you should see the button to confirm closing the conversation when RBA is enabled, and it shouldn't be there when RBA is disabled.
* After closing the conversation, find the closed conversation. On that page, you should see the button to re-open the conversation when RBA is enabled, and you shouldn't see it when RBA is disbled.
* As a bonus, run the command `kubectl get secret uaa-secret -o json --namespace=<YOUR_NAMESPACE> | jq -r '.data."uaa.yml"' | base64 -d` and you should see the `messages.delete` group listed among the other groups.

# Trello
[Card](https://trello.com/c/ZicKcJj8)
